### PR TITLE
Trac 53278: Fixes RSS Widget infinite loop when undefined URL and PHP 8+

### PR DESCRIPTION
**Pull request by hellofromtonya**
Trac ticket: https://core.trac.wordpress.org/ticket/53278

- Adds happy and unhappy path unit tests for testing the URL argument
- Applies `53278.diff` patch. Props @dd32 
- Replaces implicit loose comparison with `! empty()` ([see comment 7](https://core.trac.wordpress.org/ticket/53278#comment:7));

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/widgets/wpWidgetRss.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		$actual = ob_get_clean();
+
+		$this->assertContains( $expected, $actual );
+	}
+
+	public function data_url_happy_path() {
+		return array(
+			'when url is given' => array(
+				'url' => 'http://wordpress.org/news/feed/',
+				'<section id="widget_rss-5" class="widget widget_rss"><h2><a class="rsswidget" href="http://wordpress.org/news/feed/">',
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/widgets/wpWidgetRss.php:95]:
```suggestion
				'url' => 'https://wordpress.org/news/feed/',
				'<section id="widget_rss-5" class="widget widget_rss"><h2><a class="rsswidget" href="https://wordpress.org/news/feed/">',
```

**Review by peterwilsoncc — COMMENTED** [review]
Thanks Tonya, just a minor comment about avoiding the HTTP request in the tests. I've also suggested switching the url to https as w.org redirects.

File: tests/phpunit/tests/widgets/wpWidgetRss.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+	/**
+	 * @ticket 53278
+	 * @covers WP_Widget_RSS::widget
+	 * @dataProvider data_url_happy_path
+	 *
+	 * @param mixed  $url      URL argument.
+	 * @param string $expected Expected output.
+	 */
+	public function test_url_happy_path( $url, $expected ) {
```
[the remark this answers is not in the document]
Comment (hellofromtonya) [reply tests/phpunit/tests/widgets/wpWidgetRss.php:67]: Aha yes, mocking the response is much better for test environment performance.  Great suggestion @peterwilsoncc 🎉 

Applied in PR #1359 commit  https://github.com/WordPress/wordpress-develop/pull/1359/commits/afd358ce80840867be263ac5482b8ac9d3d0f3dd

File: tests/phpunit/tests/widgets/wpWidgetRss.php (hunk 1 above)
Comment (hellofromtonya) [reply tests/phpunit/tests/widgets/wpWidgetRss.php:95]: Applied in PR #1359 commit  https://github.com/WordPress/wordpress-develop/pull/1359/commits/afd358ce80840867be263ac5482b8ac9d3d0f3dd

**Comment by hellofromtonya** [comment]
Source code merged in changeset https://core.trac.wordpress.org/changeset/51107.

The tests and improvements to the tests is moved to PR #1359.

**Pull request closed without merging**